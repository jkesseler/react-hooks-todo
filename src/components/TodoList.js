/* eslint-disable react/no-array-index-key */
import React, { memo, useContext } from 'react';
import { List, Paper } from '@material-ui/core';
import TodoListItem from './TodoListItem';

import Store from '../store';


const TodoList = memo(() => {
  const { state, dispatch } = useContext(Store);
  const { todos } = state;

  if (todos && todos.length > 0) {
    return (
      <Paper style={{ margin: 16 }}>
        <List style={{ overflow: 'scroll' }}>
          {todos.map((todo, idx) => (
            <TodoListItem
              {...todo}
              key={idx}
              divider={idx !== todo.length - 1}
              onButtonClick={() => dispatch({ type: 'REMOVE', payload: todo })}
              onCheckboxToggle={() => dispatch({ type: 'CHECKED', payload: todo })}
            />
          ))}
        </List>
      </Paper>
    );
  }
  return null;
});

export default TodoList;
