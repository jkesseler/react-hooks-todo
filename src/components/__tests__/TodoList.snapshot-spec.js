import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { NIL as NIL_UUID } from 'uuid';
import { TodoList } from '../TodoList';

const componentProps = {
  todos: [{
    id: NIL_UUID,
    text: 'test',
  }],
  toggleTodo: () => {},
  removeTodo: () => {},
};


describe('TodoList', () => {
  it('renders without crashing', () => {
    const component = shallow(<TodoList {...componentProps} />);
    expect(shallowToJson(component)).toMatchSnapshot();
  });
});
