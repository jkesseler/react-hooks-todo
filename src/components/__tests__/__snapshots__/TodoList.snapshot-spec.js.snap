// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoList renders without crashing 1`] = `
<WithStyles(ForwardRef(Paper))>
  <WithStyles(ForwardRef(List))
    style={
      Object {
        "overflow": "scroll",
      }
    }
  >
    <Memo(TodoListItem)
      divider={true}
      id="00000000-0000-0000-0000-000000000000"
      key="0"
      onButtonClick={[Function]}
      onCheckboxToggle={[Function]}
      text="test"
    />
  </WithStyles(ForwardRef(List))>
</WithStyles(ForwardRef(Paper))>
`;
